test/hotspot/jtreg/runtime/Safepoint/TestAbortVMOnSafepointTimeout.java

```diff
@@ -23,79 +24,57 @@
-            return;
-        }
-
-        testWith(500, 500);
-    }
-
-    static int test_loop(int x) {
-        int sum = 0;
-        if (x != 0) {
-            // Long running loop without safepoint.
-            for (int y = 1; y < Integer.MAX_VALUE; ++y) {
-                if (y % x == 0) ++sum;
-            }
-        }
-        return sum;
-    }
-
-    public static void testWith(int sfpt_interval, int timeout_delay) throws Exception {
-        // -XX:-UseCountedLoopSafepoints - is used to prevent the loop
-        // in test_loop() to poll for safepoints.
-        // -XX:LoopStripMiningIter=0 and -XX:LoopUnrollLimit=0 - are
-        // used to prevent optimizations over the loop in test_loop()
-        // since we actually want it to provoke a safepoint timeout.
-        // -XX:-UseBiasedLocking - is used to prevent biased locking
-        // handshakes from changing the timing of this test.
         ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xbootclasspath/a:.",
                 "-XX:+UnlockDiagnosticVMOptions",
-                "-XX:-UseBiasedLocking",
+                "-XX:+WhiteBoxAPI",
                 "-XX:+SafepointTimeout",
                 "-XX:+SafepointALot",
                 "-XX:+AbortVMOnSafepointTimeout",
-                "-XX:SafepointTimeoutDelay=" + timeout_delay,
-                "-XX:GuaranteedSafepointInterval=" + sfpt_interval,
-                "-XX:-TieredCompilation",
-                "-XX:-UseCountedLoopSafepoints",
-                "-XX:LoopStripMiningIter=0",
-                "-XX:LoopUnrollLimit=0",
-                "-XX:CompileCommand=compileonly,TestAbortVMOnSafepointTimeout::test_loop",
-                "-Xcomp",
+                "-XX:SafepointTimeoutDelay=50",
+                "-XX:GuaranteedSafepointInterval=1",
                 "-XX:-CreateCoredumpOnCrash",
                 "-Xms64m",
-                "TestAbortVMOnSafepointTimeout",
-                "runTestLoop"
+                "TestAbortVMOnSafepointTimeout$Test",
+                "999" /* 999 is max unsafe sleep */
         );
 
         OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldMatch("Timed out while spinning to reach a safepoint.");
         if (Platform.isWindows()) {
             output.shouldMatch("Safepoint sync time longer than");
         } else {
             output.shouldMatch("SIGILL");
             if (Platform.isLinux()) {
                 output.shouldMatch("(sent by kill)");
             }
-            output.shouldMatch("TestAbortVMOnSafepointTimeout.test_loop");
         }
         output.shouldNotHaveExitValue(0);
     }
+
+    public static class Test {
+        public static void main(String[] args) throws Exception {
+            Integer waitTime = Integer.parseInt(args[0]);
+            WhiteBox wb = WhiteBox.getWhiteBox();
+            // While no safepoint timeout.
```

Comment:
Perhaps: // Loop here to cause a safepoint timeout.

Reply:
Fixed